core/src/main/scala/org/apache/spark/deploy/PythonRunner.scala

```diff
@@ -85,6 +85,7 @@ object PythonRunner {
     // pass conf spark.pyspark.python to python process, the only way to pass info to
     // python process is through environment variable.
     sparkConf.get(PYSPARK_PYTHON).foreach(env.put("PYSPARK_PYTHON", _))
+    env.put("PYTHONHASHSEED", sys.env.getOrElse("PYTHONHASHSEED", "0"))
```

Comment:
I'd rather do `sys.env.get("PYTHONHASHSEED").foreach { ... }` to avoid polluting the environment when not needed. (Also to avoid hardcoding the default in multiple places.)